pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/MultiStageBrokerRequestHandler.java

```diff
@@ -181,11 +184,57 @@ private BrokerResponseNative handleRequest(long requestId, String query,
         sqlNodeAndOptions.getParseTimeNs() + (executionEndTimeNs - compilationStartTimeNs));
     brokerResponse.setTimeUsedMs(totalTimeMs);
     brokerResponse.setResultTable(queryResults);
+
+    attachMetadataToResponse(stats, brokerResponse);
+
     requestContext.setQueryProcessingTime(totalTimeMs);
     augmentStatistics(requestContext, brokerResponse);
     return brokerResponse;
   }
 
+  private void attachMetadataToResponse(Map<String, String> stats, BrokerResponseNative brokerResponse) {
```

Comment:
assume this is copied from QueryScheduler? it is possible to add a comment/todo to indicate we want to consolidate later?